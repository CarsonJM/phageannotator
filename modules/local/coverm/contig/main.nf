process COVERM_CONTIG {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/coverm:0.6.1--h1535e20_5':
        'biocontainers/coverm:0.6.1--h1535e20_5' }"

    input:
    tuple val(meta), path(fastq)
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("${prefix}_alignment_results.tsv")    , emit: alignment_results
    tuple val(meta), path("**_alignment_results.tsv")           , emit: bam
    path "versions.yml"                                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    coverm \\
        contig \\
        -1 ${fastq[0]} \\
        -2 ${fastq[1]} \\
        --reference ${fasta} \\
        --output-file ${prefix}_alignment_results.tsv \\
        --bam-file-cache-directory ${prefix}_bam_files/ \\
        --threads ${task.cpus} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        coverm: \$(echo \$(coverm --version 2>&1) | sed 's/^.*coverm //; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir -p ${prefix}_bam_files
    touch ${prefix}_alignment_results.tsv
    touch ${prefix}_bam_files/alignment_results.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        coverm: \$(echo \$(coverm --version 2>&1) | sed 's/^.*coverm //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
