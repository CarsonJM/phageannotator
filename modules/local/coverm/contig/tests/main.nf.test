nextflow_process {

    name "Test Process: COVERM_CONTIG"
    script "modules/local/coverm/contig/main.nf"
    process "COVERM_CONTIG"
    tag "modules_local"
    tag "coverm_contig"


    test("Test COVERM_CONTIG") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'all_samples' ], [ file(params.test_data['modules_local']['b_frag_test1_bam'], checkIfExists: true), file(params.test_data['modules_local']['b_frag_test2_bam'], checkIfExists: true) ] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
