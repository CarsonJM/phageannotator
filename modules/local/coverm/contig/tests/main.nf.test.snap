{
    "['bacteroides_fragilis']['test1_bam'] & ['bacteroides_fragilis']['test2_bam']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,f7624089f001fd3079bc43e3c3372a7e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ],
                "alignment_results": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,f7624089f001fd3079bc43e3c3372a7e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ]
            }
        ],
        "timestamp": "2023-11-13T11:52:17.129290553"
    },
    "['bacteroides_fragilis']['test1_bam'] & ['bacteroides_fragilis']['test2_bam'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ],
                "alignment_results": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3f5dd63588c677b43b85a327268cd825"
                ]
            }
        ],
        "timestamp": "2024-01-17T22:32:27.21624823"
    }
}